rad/monadic/diff-remote.rad

```diff
@@ -0,0 +1,226 @@
+(import prelude/basic :unqualified)
+(import prelude/bool :unqualified)
+(import prelude/dict :unqualified)
+(import prelude/set :as 'set)
+(import prelude/ref :unqualified)
+(import prelude/validation :as 'validator)
+(import prelude/patterns :unqualified)
+(import prelude/util :unqualified)
+(import prelude/lens :unqualified)
+
+(def chain-id
+  "The name of this chain."
+  "monadic/radicle/diffs")
+
+(def diffs
+  "DIFFs are stored in a dict in a ref, keyed by diff-number."
+  (ref {}))
+
+(def diff-counter
+  "We use a stateful counter for giving diff numbers."
+  (make-counter))
+
+(def used-nonces
+  "We keep track of the UUIDs which have been used as nonces in inputs."
+  (ref set/empty))
+
+(def unused-nonce?
```

Comment:
As a note, when we merge the project stuff we should remove this since it's been moved into a library

Reply:
:+1: